test/test_other.py

```diff
@@ -9974,6 +9974,23 @@ def compile(flags):
     compile(['-sMIN_SAFARI_VERSION=140100', '-mbulk-memory'])
     verify_features_sec_linked('nontrapping-fptoint', False)
 
+  def test_no_bulk_memory(self):
+    # The test_wasm_features test (above) uses the feature section to confirm
+    # if a feuture is present, but that doesn't work in optimizing builds
+    # since we strip the feature section in release builds.
+    # This test confirms that no DATACOUNT section is present in the final
+    # binary.
+
+    def has_data_count(filename):
+      with webassembly.Module(filename) as wasm:
+        return wasm.get_section(webassembly.SecType.DATACOUNT)
+
+    self.emcc('hello_world.c', ['-O3', '-o', 'bulk.js'])
+    self.assertTrue(has_data_count('bulk.wasm'))
+
+    self.emcc('hello_world.c', ['-O3', '-o', 'nobulk.js', '-mno-bulk-memory', '-mno-bulk-memory-opt'])
```

Comment:
I wonder if it might make sense to test the `'-mno-bulk-memory'` and `'-mno-bulk-memory-opt'` separately?

IIUC `'-mno-bulk-memory'` implies `'-mno-bulk-memory-opt'`?

So `self.emcc('hello_world.c', ['-O3', '-o', 'nobulk.js', '-mno-bulk-memory']` should not get bulk memory either?

Reply:
> IIUC `'-mno-bulk-memory'` implies `'-mno-bulk-memory-opt'`?

From the LLVM POV, I'm not sure it does sadly.